### PR TITLE
fix: parse providerId URL param safely as bigint to prevent overflow

### LINKED ISSUE #182: debug params should convert safely to bigint from string
see https://github.com/filecoin-project/filecoin-pin-website/pull/181#discussion_r3234937177

> executeUpload/upload results elsewhere in the app treat provider IDs as bigint (e.g. use-upload-orchestration handles providerId: bigint | string), but here the URL providerId is parsed into a JS number and then converted with BigInt(number). That conversion will silently produce the wrong bigint if the ID exceeds Number.MAX_SAFE_INTEGER. Prefer parsing the raw query param as a bigint (with validation like > 0n) and only then passing providerIds to executeUpload, or at minimum enforce Number.isSafeInteger before converting.

We should just use BigInts <-> string for providerId

### PR DESCRIPTION
**Pull request by andi-nugroho**
### Description
This PR resolves issue #182 by ensuring that the `providerId` URL query parameter is parsed safely and directly into a `BigInt`, eliminating the intermediate conversion into a standard JavaScript `number`. 

Previously, parsing the raw string via standard JS numbers caused silent precision loss and unexpected behavior for Filecoin provider IDs that exceed `Number.MAX_SAFE_INTEGER` (9007199254740991).

### Changes
* **`src/utils/debug-params.ts`**: Introduced `parseQueryParamBigInt` utility to handle input trimming, decimal-only digit validation (`/^\d+$/`), direct `BigInt()` invocation, and `> 0n` verification. It gracefully degrades to `null` on empty or malformed parameters without crashing the application.
* **`src/hooks/use-filecoin-upload.ts`**: Updated hook to forward the parsed `bigint | null` parameter straight to `executeUpload`.
* **`src/hooks/use-data-set-manager.ts`**: Adjusted internal type references to match `bigint | null`.
* **`src/utils/debug-params.test.ts`**: Created new comprehensive unit tests validating standard behavior, empty params, malformed inputs, and edge-case provider IDs exceeding `Number.MAX_SAFE_INTEGER`.
* **`src/hooks/use-filecoin-upload.test.ts`**: Adjusted existing mock specifications to fit the new types properly.

### Verification Done
* All formatting rules strictly enforced specifically to the changed files.
* Ran local verification test suite (`npm run test`) successfully passing all 29 tests.

### CODE REVIEW

File: src/utils/debug-params.test.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest'
+import { getDebugParams } from './debug-params.ts'
+
+const UNSAFE_INTEGER_PROVIDER_ID = '9007199254740993'
+
+describe('getDebugParams providerId', () => {
```
Comment (SgtPooki) [inline src/utils/debug-params.test.ts]:
```suggestion
describe('getDebugParams', () => {
```

**Review by SgtPooki — APPROVED** [review]
just one thing to fix here.

**Comment by BigLep** [comment]
@SgtPooki : ok to merge?
